src/main/kotlin/com/mparticle/kits/AppboyKit.kt

```diff
@@ -216,6 +254,42 @@ open class AppboyKit : KitIntegration(), AttributeListener, CommerceListener,
 
     override fun logEvent(event: CommerceEvent): List<ReportingMessage> {
         val messages: MutableList<ReportingMessage> = LinkedList()
+
+        //For CommerceEvent, Event Name is not required to generate hash. So, it will be always null.
+        event.products?.forEach {
+            it.customAttributes?.let {
+                changeUserArray(it, CommerceEventUtils.getEventType(event), null, true)
+            }
+        }
+        event.transactionAttributes?.let {
+            val map = mapOf(
+                "Affiliation" to it.affiliation,
```

Comment:
you should be using the constants that are in here: https://github.com/mParticle/mparticle-android-sdk/blob/main/android-kit-base/src/main/java/com/mparticle/kits/CommerceEventUtils.java#L324

and you are missing quite a few constants/mappings - look at every single call to "hashForFiltering()" in these methods - that is a potential constant that you should likely support:
-  `filterCommerceEvent`
- `filterCommerceEntities`
- `filterCommerceEntityAttributes`
- `filterCommerceEventAttributes`.

I'd encourage you how to think about how to move this logic into that class so that we have complicated business logic close together vs separated. maybe a function in KitConfiguration or in CommerceEventUtils which accepts in a CommerceEvent and returns a map of <hash, value> - and then you can just use that map in the kit.